internal/fsutil/dir_unix.go

```
@@ -0,0 +1,73 @@
+	}
+	defer func() { _ = unix.Close(parentFD) }()
+
+	if err := unix.Mkdirat(parentFD, name, privateDirMode); err != nil && !errors.Is(err, unix.EEXIST) {
+		return fmt.Errorf(messages.FsutilPrivateDirCreateFmt, path, err)
+	}
+
+	var st unix.Stat_t
+	if err := unix.Fstatat(parentFD, name, &st, unix.AT_SYMLINK_NOFOLLOW); err != nil {
+		return fmt.Errorf(messages.FsutilPrivateDirStatFmt, path, err)
+	}
+	if !isPrivateDirMode(uint32(st.Mode)) {
```

golangci-lint (2.12.2)

[error] unnecessary conversion

(unconvert)